### PR TITLE
Change the quarkus-kotlin extension status to stable

### PR DESCRIPTION
**Pull request by aloubyansky**
Following the discussion https://groups.google.com/g/quarkus-dev/c/Yz3BriSfsf4/m/HVFLNfPtCAAJ?utm_medium=email&utm_source=footer

It doesn't seem like there are major issues to fix to earn the `stable` status for `quarkus-kotlin`. Any objections?

### CODE REVIEW

File: docs/src/main/asciidoc/kotlin.adoc [hunk 1]
```diff
@@ -5,7 +5,6 @@ https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 ////
 = Using Kotlin
 
-:extension-status: preview
```
Comment (gsmet) [inline docs/src/main/asciidoc/kotlin.adoc:8]: Please also remove `include::{includes}/extension-status.adoc[]` a few lines below.

**Review by gsmet — DISMISSED** [review]

**Review by glefloch — APPROVED** [review]
Looks good to me

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by kenyee** [comment]
Also would be nice to have someplace where there's docs for the caveats mentioned in the the dev thread for Kotlin support.
e.g. panache-reactive module doesn't support Kotlin, Kotlin coroutines aren't supported everywhere, etc.

File: docs/src/main/asciidoc/kotlin.adoc (hunk 1 above)
Comment (aloubyansky) [reply docs/src/main/asciidoc/kotlin.adoc:8]: Thanks, done.

Comment (gastaldi) [reply docs/src/main/asciidoc/kotlin.adoc:8]: @gsmet I see you asked the same in https://github.com/quarkusio/quarkus/pull/26709. There is a nice message when the extension status is stable, what's the reason behind removing it? 

image

**Comment by evanchooly** [comment]
we should probably also "promote" the other kotlin modules.  

- `mongodb-panache-kotlin` is listed as `stable` already 
- `hibernate-orm-panache-kotlin` is still `experimental`.  
- `quarkus-resteasy-reactive-kotlin` doesn't have anything in it's `quarkus-extension.yaml` at all.  I don't know what that means.
- `resteasy-reactive-kotlin-serialization` is listed as `experimental`

**Review by evanchooly — CHANGES_REQUESTED** [review]
we should probably also "promote" the other kotlin modules.

  - mongodb-panache-kotlin is listed as stable already
  - hibernate-orm-panache-kotlin is still experimental.
  - quarkus-resteasy-reactive-kotlin doesn't have anything in it's quarkus-extension.yaml at all. I don't know what that means.
  - resteasy-reactive-kotlin-serialization is listed as experimental

**Comment by aloubyansky** [comment]
> we should probably also "promote" the other kotlin modules.
> 
> * mongodb-panache-kotlin is listed as stable already
> * hibernate-orm-panache-kotlin is still experimental.
> * quarkus-resteasy-reactive-kotlin doesn't have anything in it's quarkus-extension.yaml at all. I don't know what that means.
> * resteasy-reactive-kotlin-serialization is listed as experimental

Just to clarify, @evanchooly you would like to make them all `stable`?

`quarkus-resteasy-reactive-kotlin` is `unlisted` which means it's not exposed to users for explicit selection. It's activated automagically when both `quarkus-resteasy-reactive` and `quarkus-kotlin` are found among the project's dependencies.

**Comment by aloubyansky** [comment]
Changed the statuses of `hibernate-orm-panache-kotlin` and `resteasy-reactive-kotlin-serialization` to `stable`. I left `quarkus-resteasy-reactive-kotlin` as-is, given that we aren't currently displaying unlisted extensions to users and if a status is missing, I guess we could assume it's stable, although we may want to discuss that.

File: docs/src/main/asciidoc/kotlin.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/kotlin.adoc:8]: @gastaldi we always only included the note in non-stable extensions so I want us to stay consistent.

**Comment by gsmet** [comment]
@n1hility I think you should probably be involved in this discussion too.

**Comment by n1hility** [comment]
@evanchooly do you expect any major changes to the extensions in this list? I just want to make sure that we are at a stage where we feel good about the APIs and they are not under flux.

**Comment by evanchooly** [comment]
I don't.   All the panache bits are driven by common interfaces and the others don't have much or any user facing interfaces.

**Comment by n1hility** [comment]
Ok then I am +1 sans any extensions we have doubts about.

**Comment by gsmet** [comment]
@evanchooly I let you have a look to the latest changes and merge if ready.

**Review by evanchooly — APPROVED** [review]

**Pull request merged**